### ISSUE TITLE
The wazuh-monitord daemon increased its amount of disk read

### ISSUE THREAD

**Issue by mauromalara** [wazuh/wazuh#19384]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.5.3-rc2 | wazuh-monitord | Manager | - | - |

## Description

Compared with the [last test session](https://github.com/wazuh/wazuh/issues/19186) (4.5.3-rc1) `wazuh-monitord` seems to increase its amount of disk read during the testing session.

### Evidence

**`4.5.3-rc1` (last test session)**

<img src="https://camo.githubusercontent.com/b593203f6f5f984946e776cb348a9f0fb6f05b4fe4094845c8af9a8a3f5f1fbf/68747470733a2f2f63692e77617a75682e636f6d2f646174612f546573745f7374726573732f7072652d72656c656173652f342e352e332f42343333312d333630306d2f42343333315f6d616e616765725f63656e746f732f706c6f74732f6d6f6e69746f722d6d616e616765722d546573745f7374726573735f42343333315f6d616e616765722d7072652d72656c656173655f4469736b5f526561642e706e67">

**`4.5.3-rc2` (current test session)**

<img src="https://camo.githubusercontent.com/0299b63e7385f075feb78f111af0e2acdd05aec672af0a72e5e5257b2c816310/68747470733a2f2f63692e77617a75682e636f6d2f646174612f546573745f7374726573732f7072652d72656c656173652f342e352e332f42343337372d333630306d2f42343337375f6d616e616765725f63656e746f732f706c6f74732f6d6f6e69746f722d6d616e616765722d546573745f7374726573735f42343337375f6d616e616765722d7072652d72656c656173655f4469736b5f526561642e706e67">

- The configuration hasn’t changed since the last testing session.
- [manager_logs_and_config.zip](https://ci.wazuh.com/data/Test_stress/pre-release/4.5.3/B4377-3600m/B4377_manager_centos/logs/ossec_Test_stress_B4377_manager_2023-10-01.zip)
- CSVs:
  - [monitor-manager-Test_stress_B4377_manager-pre-release.csv](https://ci.wazuh.com/data/Test_stress/pre-release/4.5.3/B4377-3600m/B4377_manager_centos/data/monitor-manager-Test_stress_B4377_manager-pre-release.csv)
  - [Test_stress_B4377_manager_analysisd_events.csv](https://ci.wazuh.com/data/Test_stress/pre-release/4.5.3/B4377-3600m/B4377_manager_centos/data/Test_stress_B4377_manager_analysisd_events.csv)
  - [Test_stress_B4377_manager_analysisd_state.csv](https://ci.wazuh.com/data/Test_stress/pre-release/4.5.3/B4377-3600m/B4377_manager_centos/data/Test_stress_B4377_manager_analysisd_state.csv)
  - [Test_stress_B4377_manager_remoted_state.csv](https://ci.wazuh.com/data/Test_stress/pre-release/4.5.3/B4377-3600m/B4377_manager_centos/data/Test_stress_B4377_manager_remoted_state.csv)

**Comment by cborla** [comment]
## Monitord Analysis

One of the tasks that monitord performs is to rotate and compress the logs that are generated. The time of log generation is at 00:00:00 + a configurable offset in internal_options `monitord.day_wait`,  

The configuration used for the test is as follows:

```
# Monitord day_wait. Amount of seconds to wait before rotating/compressing/signing [0..600]
# the files.
monitord.day_wait=10

# Monitord compress. (0=do not compress, 1=compress)
monitord.compress=1

# Monitord sign. (0=do not sign, 1=sign)
monitord.sign=1

# Monitord monitor_agents. (0=do not monitor, 1=monitor)
monitord.monitor_agents=1

# Rotate plain and JSON logs daily. (0=no, 1=yes)
monitord.rotate_log=1

# Days to keep old ossec.log files [0..500]
monitord.keep_log_days=31

# Size of internal log files to rotate them (Megabytes) [0..4096]
monitord.size_rotate=512

# Maximum number of rotations per day for internal logs [1..256]
monitord.daily_rotations=12

# Number of minutes for deleting a disconnected agent [0..9600]. (0=disabled)
monitord.delete_old_agents=0
```

The clear step you see in the graphs is at 00:00:00, whether the step is inside or outside the stress zone is completely conditioned to the time at which the test starts. It would be convenient to try to launch the tests so that the results are as similar as possible.

| Version | Build | Disk read Graph (MB)| size `/var/ossec/logs/wazuh` (KB) |
|---|---|---|---|
| 4.5.3 RC-1 | 4331 | 0.60 | 644 |  
| 4.5.3 RC-2 | 4377 | 0.75 | 792 |

The difference of approximately 150 is conditional on the size of the rotating/compressing/signing files, looking at the size of the logs rotated in  `/var/ossec/logs/wazuh` we found similar differences to those plotted.

In conclusion, we understand that there is no problem in monitord, only different conditions that generated differences in disk reading.

**Comment by Dwordcito** [comment]
LGTM, the explanation makes sense @cborla
